### PR TITLE
Set the event date to the largest date when deduplicating

### PR DESCRIPTION
**Pull request by srijken**

### CODE REVIEW

**Comment by niemyjski** [comment]
Looks like there is conflicts now with my updates :(. There seems to be a race condition that is causing multiple issues that I've been running into. I added a lock and it seemed to help but something is causing the same event to have different hashcodes.

File: Source/Shared/Plugins/Default/1010_DuplicateCheckerPlugin.cs [hunk 1]
```diff
@@ -102,6 +104,13 @@ private class MergedEvent {
                 _context.Event.Count = _count;
                 _context.Resolver.GetEventQueue().Enqueue(_context.Event);
             }
+
+            public void UpdateDate(DateTimeOffset date) {
+                lock (_lock) {
```
Comment (niemyjski) [inline Source/Shared/Plugins/Default/1010_DuplicateCheckerPlugin.cs]: Would we still need this lock if we are locking on timer run and enqueue?

Comment (srijken) [inline Source/Shared/Plugins/Default/1010_DuplicateCheckerPlugin.cs]: Well, the same could be said for Interlocked.Increment in IncrementCount()?

Comment (niemyjski) [inline Source/Shared/Plugins/Default/1010_DuplicateCheckerPlugin.cs]: I don't think we need that lock there upon a second look. The outside is wrapped in it. I don't see anything wrong with interlocked.Increment (atomic)

**Pull request merged**